WalletWasabi/Blockchain/Keys/HdPubKey.cs

```diff
@@ -28,9 +28,10 @@ public HdPubKey(PubKey pubKey, KeyPath fullKeyPath, SmartLabel label, KeyState k
 		KeyState = keyState;
 
 		P2pkScript = PubKey.ScriptPubKey;
-		P2pkhScript = PubKey.Hash.ScriptPubKey;
+		P2pkhScript = PubKey.GetScriptPubKey(ScriptPubKeyType.Legacy);
```

Comment:
Not related to the review - just a question, how much does it take (time) to get this? Would it make sense to use `Lazy`?

Reply:
Nothing. It is implemented with something like `if (Lageacy) then return Hash`

> Would it make sense to use Lazy?

No, absolutely not.